tests/compiler.js

```diff
@@ -1570,5 +1570,41 @@
 
             finish(done);
         });
+
+        it('should get right value when macro parameter conflict with global macro name', function(done) {
+            render(
+                '{# macro1 and macro2 definition #}' +
+                '{% macro macro1() %}' +
+                '{% endmacro %}' +
+                '' +
+                '{% macro macro2(macro1="default") %}' +
+                '{{macro1}}' +
+                '{% endmacro %}' +
+                '' +
+                '{# calling macro2 #}' +
+                '{{macro2("this should be outputted") }}', {}, {}, function(err, res) {
+                    expect(res.trim()).to.eql('this should be outputted');
+            });
+
+            finish(done);
+        });
+
+        it('should get right value when macro include macro', function(done) {
+            render(
+                '{# macro1 and macro2 definition #}' +
+                '{% macro macro1() %} foo' +
+                '{% endmacro %}' +
+                '' +
+                '{% macro macro2(text="default") %}' +
+                '{{macro1()}}' +
+                '{% endmacro %}' +
+                '' +
+                '{# calling macro2 #}' +
+                '{{macro2("this should be outputted") }}', {}, {}, function(err, res) {
```

Comment:
this test output asset to equal `foo`. my last post code is 

```
{# macro1 and macro2 definition #}
{% macro macro1() %}
{% endmacro %}

{% macro macro2(text="default") %}
{{macro1}}
{% endmacro %}

{# calling macro2 #}
{{macro2("this should be outputted") }}
```

in this case, output will be the `macro1` source code. But this output have no problem, because `{{macro1}}` should output this function toString. so I delete that post, which is mistake.

After I change it to `{{macro1}}`.